service/src/test/java/datawave/microservice/authorization/config/CachingConfigurerTest.java

```diff
@@ -0,0 +1,250 @@
+    
+    @Autowired
+    private TestCache testCache;
+    
+    @BeforeEach
+    public void setup() {
+        TestCache.putError = false;
+        TestCache.evictError = false;
+        TestCache.clearError = false;
+    }
+    
+    @Test
+    public void testCacheGetError() {
+        // Cache some stuff...
+        testService.get("foo");
+        testService.get("bar");
+        
+        assertEquals("foo", testCache.get("foo").get());
+        assertEquals("bar", testCache.get("bar").get());
+        try {
+            testService.get(ERROR_KEY);
+        } catch (RuntimeException e) {
+            assertEquals("Simulated error for testing.", e.getMessage());
+        }
+        assertTrue(testCache.getNativeCache().isEmpty(), "Cache should have been cleared after a get error, but it wasn't.");
+    }
+    
+    @Test
+    public void testCachePutError() {
+        TestCache.putError = true;
+        
+        Object response = testService.get("putFailed");
+        assertEquals("putFailed", response);
+        
+        assertNull(testCache.get("putFailed"), "Object should not have been cached");
+    }
+    
+    @Test
+    public void testCacheEvictError() {
+        TestCache.evictError = true;
+        
+        testService.get("foo");
+        assertEquals("foo", testCache.get("foo").get());
+        
+        testService.evict("foo");
+        
+        // Make sure it is still cached, but we got here if the evict call didn't fully fail
+        assertEquals("foo", testCache.get("foo").get());
+    }
+    
+    @Test
+    public void testCacheClearError() {
+        TestCache.clearError = true;
+        
+        testService.get("foo");
+        assertEquals("foo", testCache.get("foo").get());
+        
+        testService.clear();
+        
+        // Make sure it is still cached, but we got here if the evict call didn't fully fail
+        assertEquals("foo", testCache.get("foo").get());
+    }
+    
+    // @EnableCaching
+    @ImportAutoConfiguration({RefreshAutoConfiguration.class})
+    @AutoConfigureCache(cacheProvider = CacheType.HAZELCAST)
+    @ComponentScan(basePackages = "datawave.microservice")
+    @Profile("CachingConfigurerTest")
+    @Configuration
+    public static class CachingConfiguration {
+        @Bean
+        public CachedDatawaveUserService cachedDatawaveUserService(CacheManager cacheManager, Function<CacheManager,CacheInspector> cacheInspectorFactory) {
+            return new AuthorizationTestUserService(Collections.EMPTY_MAP, true);
+        }
+        
+        @Bean
+        public HazelcastInstance testHazelcastInstance() {
+            Config config = new Config();
+            config.setClusterName(UUID.randomUUID().toString());
+            config.getNetworkConfig().getJoin().getMulticastConfig().setEnabled(false);
+            return Hazelcast.newHazelcastInstance(config);
+        }
+        
+        @Bean
+        public BusProperties pusProperties() {
```

Comment:
typo

Reply:
lol, that's a fun typo.  should be fixed now.